redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,69 @@
+package redisxtest
+
+import (
+	"context"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// MockRedisCluster wraps a local version of redis
+type MockRedisCluster struct {
+	redisCluster *miniredis.Miniredis
+}
+
+func NewMockRedisCluster() (mockRedisCluster MockRedisCluster, teardown func(), err error) {
+	redisCluster, err := miniredis.Run()
+	if err != nil {
+		return MockRedisCluster{}, nil, err
+	}
+
+	mockRedisCluster = MockRedisCluster{
+		redisCluster: redisCluster,
+	}
+
+	teardown = func() {
+		mockRedisCluster.Close()
+	}
```

Comment:
this is a very standard closing pattern in go (especially when you make `MockRedisCluster` to implement `io.Closer`), so I would say returning a `teardown` is unnecessary, caller could just do:
```go
cluster, err := NewMockRedisCluster()
if err != nil { ... }
defer cluster.Close()
```